src/wanderer.ts

```diff
@@ -206,6 +219,52 @@ function guzzlrAbandonQuest() {
   }
 }
 
+export function averageYrValue(location: Location, debug = false) {
+  const monsters = Object.keys(getLocationMonsters(location))
+    .map((m) => toMonster(m))
+    .filter((m) => !["LUCKY", "ULTRARARE", "BOSS"].some((s) => m.attributes.includes(s)));
+
+  if (monsters.length === 0) {
+    return 0;
+  } else {
+    return (
+      sumNumbers(
+        monsters.map((m) => {
+          const items = itemDropsArray(m)
+            .filter((drop) => ["", "n"].includes(drop.type))
+            .map((drop) => garboValue(drop.drop));
+
+          if (debug) {
+            print(`${m}: ${items.join(",")}`);
+          }
+          return sumNumbers(
+            itemDropsArray(m)
+              .filter((drop) => ["", "n"].includes(drop.type))
+              .map((drop) => garboValue(drop.drop))
+          );
+        })
+      ) / monsters.length
+    );
+  }
+}
+
+export function bestYrLocation(): { location: Location; value: number } {
+  const validLocations = $locations``
+    .filter(canAdventureOrUnlock)
+    .filter((l) => canWander(l, "yellow ray"))
+    .map((location) => {
+      return { location, value: averageYrValue(location) };
+    })
+    .filter((value) => value.value > 0);
+  if (validLocations.length > 0) {
+    const res = validLocations.sort((a, b) => b.value - a.value)[3];
```

Comment:
let's use a reduce instead of a sort.

Why are we looking at the fourth best? Is that because we haven't hacked the bad zones yet?

Reply:
We can get rid of the [3] now that we've nixed the clan dungeon